---
title: "Lab 14 Homework"
author: "Please Add Your Name Here"
date: "`r Sys.Date()`"
output: 
  html_document: 
    keep_md: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Instructions
Answer the following questions and complete the exercises in RMarkdown. Please embed all of your code and push your final work to your repository. Your final lab report should be organized, clean, and run free from errors. Remember, you must remove the `#` for the included code chunks to run. Be sure to add your name to the author header above. For any included plots, make sure they are clearly labeled. You are free to use any plot type that you feel best communicates the results of your analysis.  

Make sure to use the formatting conventions of RMarkdown to make your report neat and clean!  

## Load the libraries
```{r message=FALSE, warning=FALSE}
library(qtl)
library(qtlcharts)
library(tidyverse)
library(ggmap)
library(naniar)
library(janitor)
```

**1. We have a satellite collars on a number of different individuals and want to be able to quickly look at all of their recent movements at once. Please load all the data files from `us_individual_collar_data` and use for loop to create plots for all different individuals of the path they move on longitude and latitude.**
```{r}
us_collars <- list.files("data/us_individual_collar_data", pattern = "collar-data", full.names = TRUE)
us_collars
```
```{r}
i <- 1
for (i in 1:length(us_collars)){
  data0 <- as.data.frame(read_csv(us_collars[i]))
  print(
    ggplot(data0, aes(x=long,y=lat))+
      geom_path()+
      geom_point()+
      labs(title = us_collars[i], x = "longitude", y = "latitude")
  )
}
```

**2. Please load all the data files from `us_individual_collar_data` and combine all data into one data frame. Create a summary to show what is the maximum and minimum of recorded data points on longitude and latitude.**
```{r}
collar_list <- lapply(us_collars, read_csv)
us_collars_combined <- bind_rows(collar_list)
us_collars_combined %>% 
  summarise(max_lat=max(lat),
            min_lat=min(lat),
            max_long=max(long),
            min_long=min(long))
```

**3. Use the range of the latitude and longitude from Q2 to build an appropriate bounding box for your map and load a map from `stamen` in a terrain style projection and display the map. Then, build a final map that overlays the recorded path from Q1.**
```{r}
lat <- c(26.6116, 41.58802)
long <- c(-122.6017	, -106.3343	)
bbox <- make_bbox(long, lat, f = 0.5)
map <- get_map(bbox, maptype = "terrain", source = "stamen")
ggmap(map)
```
```{r}
for (i in 1:length(us_collars)){
  data0 <- as.data.frame(read_csv(us_collars[i]))
  print(
    ggmap(map)+
      geom_path(data = data0, aes(long,lat))+
      geom_point(data = data0, aes(long,lat))+
      labs(title = us_collars[i], x = "longitude", y = "latitude")
  )
}
```
## Load the Data
We will use the data from an experiment on hypertension in the mouse [Sugiyama et al., Genomics 71:70-77, 2001](https://www.sciencedirect.com/science/article/pii/S0888754300964012?via%3Dihub)
```{r}
#?hyper
data(hyper)
```

**4. Create a summary of the hypertension data. How many individuals and phenotypes are included in this data set? How many gene markers and chromosomes are included in this data set? Please create a table to show the number of markers on each chromosome.**
```{r}
head(hyper$pheno)
```

```{r}
summary(hyper)
```
```{r}
nind(hyper)    
nphe(hyper)   
nchr(hyper)  
totmar(hyper) 
nmar(hyper)    
```

```{r}
plot(hyper)
```

**5. Please make an interactive genetic map of markers for the hypertension data.**
```{r}
iplotMap(hyper)
```

**6. Make a plot shows the pattern of missing genotype data in the hypertension dataset. Please reorder the recorded individuals according to their blood pressure phenotypes. Is there a specific pattern of missing genotype? Please explain it.**
```{r}
plotMissing(hyper, main = "", reorder = 1)
```

**7. Based on your answer from previous question, you probably noticed that there are gene markers without data. Please use the function `drop.nullmarkers` to remove markers that have no genotype data. After this, make a new summary to show the number of markers on each chromosome. How many gene markers were dropped? Where were the dropped markers located? Please use the data without nullmarkers for the following questions.**
```{r}
drop.nullmarkers(hyper)
hyper_dropped <- drop.nullmarkers(hyper)
summary(hyper_dropped)
```
```{r}
nmar(hyper)
```
```{r}
nmar(hyper_dropped)
```
It looks like a single 5 in the 2nd row was dropped for a 4.
**8. Please conduct single-QTL analysis and create a table to give the maximum LOD score on each chromosome based on their blood pressure phenotypes. Which gene marker has the higiest LOD score? Which chromosome contains the gene marker that has the highest LOD score? Then, creates an interactive chart with LOD curves from a genome scan for all chromosomes.**
```{r}
hyper_more_dropped <- calc.genoprob(hyper_dropped, step=1)
QTL <- scanone(hyper_more_dropped)
QTL %>% 
  summary() %>% 
  arrange(desc(lod))

```

```{r}
iplotScanone(QTL)
```
<style>
div.blue { background-color:#e6f0ff; border-radius: 5px; padding: 20px;}
</style>
<div class = "blue">

**9. Based on your genome scan results, create a table which only includes those chromosomes with LOD > 1. Creates an interactive chart with LOD curves linked to estimated QTL effects for only those chromosomes with LOD > 1. Find the gene maker with the highest LOD score and describe how does the genetype of this marker influence the individual's phenotype.**
```{r}
summary(QTL, threshold=1)
```
```{r}
iplotScanone(QTL, hyper_dropped, chr = c(1, 2, 4, 5, 6, 15))
```
</div>

**10. Please save your interactive chart from Q9 as a html file `hyper_iplotScanone.html` and make sure your upload it to your github repository with your lab14 homework as well.**
```{r}
scanoe_chart <- iplotScanone(QTL, hyper_dropped, chr = c(1, 2, 4, 5, 6, 15))
htmlwidgets::saveWidget(scanoe_chart, file = "scanoe_chart.html")
```

## Push your final code to GitHub!
Please be sure that you check the `keep md` file in the knit preferences. 